perfkitbenchmarker/linux_packages/cuda_toolkit_8.py

```diff
@@ -0,0 +1,109 @@
+# Copyright 2016 PerfKitBenchmarker Authors. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+"""Module containing CUDA toolkit 8 installation and cleanup functions."""
+
+from perfkitbenchmarker import regex_util
+
+# TODO: Test the CUDA Ubuntu 14.04 installer, and if everything works ok,
+# automatically install the correct package depending on the OS image.
+CUDA_TOOLKIT_UBUNTU = 'cuda-repo-ubuntu1604_8.0.44-1_amd64.deb'
+CUDA_TOOLKIT_UBUNTU_URL = (
+    'http://developer.download.nvidia.com/compute/cuda'
+    '/repos/ubuntu1604/x86_64/%s' % CUDA_TOOLKIT_UBUNTU)
+CUDA_TOOLKIT_INSTALL_DIR = '/usr/local/cuda'
+
+EXTRACT_CLOCK_SPEEDS_REGEX = r'(\d*).*,\s*(\d*)'
+
+
+def QueryNumberOfGpus(vm):
+  """Returns the number of Nvidia GPUs on the system"""
+  stdout, _ = vm.RemoteCommand("sudo nvidia-smi --query-gpu=count --id=0 "
+                               "--format=csv", should_log=True)
+  return int(stdout.split()[1])
+
+
+def SetGpuClockSpeed(vm, memory_clock_speed, graphics_clock_speed):
+  """Sets the K80 GPU memory and graphics clocks to the specified frequency
+     and enables persistence mode.
+
+     Note that these settings are lost after reboot.
+
+     Args:
+      vm: virtual machine to operate on
+      memory_clock_speed: desired speed of the memory clock, in MHz
+      graphics_clock_speed: desired speed of the graphics clock, in MHz
+  """
+  vm.RemoteCommand('sudo nvidia-smi -pm 1')
+  vm.RemoteCommand('sudo nvidia-smi -ac {},{}'.format(memory_clock_speed,
+                                                      graphics_clock_speed))
+
+
+def QueryGpuClockSpeed(vm, device_id):
+  """Returns the user-specified values of the memory and graphics clock
+     in MHz for the specified GPU.
+
+     Args:
+      vm: virtual machine to operate on
+      device_id: id of GPU device to query
+
+    Returns:
+      Tuple of clock speeds in MHz in the form (memory clock, graphics clock).
+  """
+  query = ("sudo nvidia-smi --query-gpu=clocks.applications.memory,"
```

Comment:
Use single quote except for doc string
Be consistent about quotes (in pkb context, single quote for everything except doc string)